source/js/components/join/join.jsx

```diff
@@ -83,9 +83,20 @@ export default class JoinUs extends React.Component {
    */
   validatesAsEmail(input) {
```

Comment:
In `processFormData(event) { ... }`, there's this line `if (email && this.validatesAsEmail(email) && consent) {`. This means even when we have validation error `this.validatesAsEmail(email)` is still true as it returns a object. 

Because of this let's change `validatesAsEmail` to the following code to make sure the type of result this method returns is consistent.

```jsx
validatesAsEmail(input) {
  if (!input) {
    return {
      valid: false,
      errorMessage: `This is a required section.`
    };
  }

  let formatValid = input.match(/[^@]+@[^.@]+(\.[^.@]+)+$/) !== null;

  if (!formatValid) {
    return {
      valid: false,
      errorMessage: `Please enter a valid email address.`
    };
  }

  return {
    valid: true
  };
}
```

To check if email is valid, we do ` this.validatesAsEmail(email).valid` instead of ` this.validatesAsEmail(email)`.